trailblazer/services/analysis_service/utils.py

```diff
@@ -0,0 +1,41 @@
+from trailblazer.constants import TrailblazerStatus
+from trailblazer.dto.analyses_response import UpdateAnalysesResponse
+from trailblazer.dto.analysis_response import AnalysisResponse
+from trailblazer.dto.summaries_response import Summary
+from trailblazer.store.models import Analysis
+
+
+def get_status_count(analyses: list[Analysis], status: TrailblazerStatus) -> int:
+    return len([a for a in analyses if a.status == status])
+
+
+def create_summary(analyses: list[Analysis]) -> Summary:
+    total: int = len(analyses)
+    delivered: int = get_status_count(analyses=analyses, status=TrailblazerStatus.COMPLETED)
+    running: int = get_status_count(analyses=analyses, status=TrailblazerStatus.RUNNING)
+    cancelled: int = get_status_count(analyses=analyses, status=TrailblazerStatus.CANCELLED)
+    failed = get_status_count(analyses=analyses, status=TrailblazerStatus.FAILED)
```

Comment:
This will work! It might be cool/concise to use `count([a.status for a in analyses])` or whatever the syntax is. Up to you though, this might just be clearer.

Reply:
I'd like that, but seems like `count` only supports direct checks on the elements in the list - not attributes on the object